pkg/util/spanlogger/spanlogger.go

```diff
@@ -94,3 +95,23 @@ func (s *SpanLogger) Error(err error) error {
 	s.Span.LogFields(otlog.Error(err))
 	return err
 }
+
+// triggerError calls s.Error if SpanLogger's log level is set to level.ErrorValue().
+func (s *SpanLogger) triggerError(kvps ...interface{}) {
+	var logAsError = false
+	loggedError := errors.New("unknown error")
+	for i := 0; i < len(kvps) - 1; i += 2 {
+		if kvps[i] == level.Key() && kvps[i+1] == level.ErrorValue() {
+			logAsError = true
+		} else if kvps[i] == "err" {
+			if err, ok := kvps[i+1].(error); ok {
+				loggedError = err
+			} else if errMsg, ok := kvps[i+1].(string); ok {
+				loggedError = errors.New(errMsg)
+			}
+		}
```

Comment:
I would simplify this, supporting only `error` even if it's not preceeded by the field name `err` (eg. may also be called differently, not sure, too many places to check);

```suggestion
		if kvps[i] == level.Key() && kvps[i+1] == level.ErrorValue() {
			logAsError = true
			continue
		}
		if err, ok := kvps[i+1].(error); ok {
				loggedError = err
		}
```